tests/test_pagination.py

```diff
@@ -615,3 +617,123 @@ def test_error_invalid_relay_cursor_field(client, data):
             }
         ],
     )
+
+
+@pytest.mark.parametrize(
+    ("args", "expected"),
+    (
+        (
+            {"cursor": "foo"},
+            CursorInfo(False, "foo", "cursor", None, None),
+        ),
+        (
+            {"cursor": "foo", "limit": 30},
+            CursorInfo(False, "foo", "cursor", 30, "limit"),
+        ),
+        (
+            {"cursor": "foo", "last": 30},
+            CursorInfo(False, "foo", "cursor", None, None),
+        ),
+        (
+            {"after": "foo"},
+            CursorInfo(False, "foo", "after", None, None),
+        ),
+        (
+            {"after": "foo", "limit": 30},
+            CursorInfo(False, "foo", "after", 30, "limit"),
+        ),
+        (
+            {"after": "foo", "first": 30},
+            CursorInfo(False, "foo", "after", 30, "first"),
+        ),
+        (
+            {"after": "foo", "first": 30, "limit": 15},
```

Comment:
suggestion (non-blocking): as discussed on Slack, it seems unnecessary to support passing both after/first and cursor/limit in the same request.

if it simplifies the implementation, i'd suggest removing